controllers/trainingjob/trainingjob_controller.go

```diff
@@ -99,6 +99,12 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 
 	switch ctx.TrainingJob.Status.TrainingJobStatus {
 	case string(sagemaker.TrainingJobStatusCompleted):
+		// If there are any debug rule job still in progress
+		for _, debugRuleJob := range ctx.TrainingJob.Status.DebugRuleEvaluationStatuses {
+			if *(debugRuleJob.RuleEvaluationStatus) == string(sagemaker.RuleEvaluationStatusInProgress) || *(debugRuleJob.RuleEvaluationStatus) == string(sagemaker.RuleEvaluationStatusStopping) {
+				return controllers.RequeueAfterInterval(r.PollInterval, nil)
+			}
+		}
 		fallthrough
 	case string(sagemaker.TrainingJobStatusFailed):
```

Comment:
Won't we want to continue requeuing if there are debug rule jobs still running in this case? Same for `TrainingJobStatusStopped`.

Reply:
Good point, I think we should keep requeuing there as well.

Reply:
Please correct me if I'm wrong @cadedaniel, but how I interpreted this is that we should be requeue-ing the training job if there are debug rules in progress, regardless of the state of the training job. It's possible that the training job has stopped, but the debug rules are still in progress. If we don't requeue, then the status of those debug rules will always appear "InProgress" in k8s.

Reply:
May be I am not communicating it properly. We do not update status of debug job asynchronously but every requeue does it automatically. In case of  training job `completion` we need to check the debug job to make sure it also reaches to its final state, but when training job is in progress requeue cycle will anyway update the debug job. So what is the issue here?

Reply:
Also since Trainingjob and debug job runs asynchronously I noticed in sagemaker even if you explicitly stop a trainingjob, debug job will stay in progress.